src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+    EmbeddedFileTypes as EmbeddedFileTypes_2023_09,
+    EmbeddedFileText as EmbeddedFileText_2023_09,
+    Action as Action_2023_09,
+    StepScript as StepScript_2023_09,
+    StepActions as StepActions_2023_09,
+)
+from openjd.model import ParameterValue
+
+from ..session import Session
+from ...log_messages import SessionActionLogKind
+
+
+from .openjd_action import OpenjdAction
+
+if TYPE_CHECKING:
+    from concurrent.futures import Future
+    from ..session import Session
+    from ..job_entities import JobAttachmentDetails, StepDetails
+
+
+logger = getLogger(__name__)
+
+
+class SyncCanceled(Exception):
+    """Exception indicating the synchronization was canceled"""
+
+    pass
+
+
+class AttachmentDownloadAction(OpenjdAction):
+    """Action to synchronize input job attachments for a AWS Deadline Cloud job
+
+    Parameters
+    ----------
+    id : str
+        The unique action identifier
+    """
+
+    _cancel: Event
+    _future: Future[None]
+    _job_attachment_details: Optional[JobAttachmentDetails]
+    _step_details: Optional[StepDetails]
+    _step_script: Optional[StepScript_2023_09]
+
+    def __init__(
+        self,
+        *,
+        id: str,
+        session_id: str,
+        job_attachment_details: Optional[JobAttachmentDetails] = None,
+        step_details: Optional[StepDetails] = None,
+    ) -> None:
+        super(AttachmentDownloadAction, self).__init__(
+            id=id,
+            action_log_kind=(
+                SessionActionLogKind.JA_SYNC
+                if step_details is None
+                else SessionActionLogKind.JA_DEP_SYNC
+            ),
+            step_id=step_details.step_id if step_details is not None else None,
+        )
+        self._cancel = Event()
+        self._job_attachment_details = job_attachment_details
+        self._step_details = step_details
+        self._logger = LoggerAdapter(OPENJD_LOG, extra={"session_id": session_id})
+
+    def set_step_script(self, manifests, path_mapping, s3_settings) -> None:
+        profile = os.environ.get("AWS_PROFILE")
+        deadline_path = os.path.join(Path(sysconfig.get_path("scripts")), "deadline")
+
+        script = "#!/usr/bin/env bash\n\n{} attachment download -m {} --path-mapping-rules {} --s3-root-uri {}  --profile {}".format(
+            deadline_path,
+            " -m ".join(quote(v) for v in manifests),
+            quote(path_mapping),
+            s3_settings.to_s3_root_uri(),
+            profile,
+        )
+
+        self._step_script = StepScript_2023_09(
+            actions=StepActions_2023_09(
+                onRun=Action_2023_09(command="{{ Task.File.AttachmentDownload }}")
+            ),
+            embeddedFiles=[
+                EmbeddedFileText_2023_09(
+                    name="AttachmentDownload",
+                    type=EmbeddedFileTypes_2023_09.TEXT,
+                    runnable=True,
+                    data=script,
+                )
+            ],
+        )
```

Comment:
I may be missing something, but this looks like it is a single command being run in this script. Do we need the embedded file at all? Or can we just invoke the command directly in the `onRun` action?

Removing `bash` from the solution removes a hidden dependency and command injection issues.

Reply:
I'd like to keep it as-is for now and convert it to python as embedded files later. That should come after the upload PR and along with other potential windows change.

Reply:
Synced offline, encountered file not found issue trying to directly invoke the python script, and having the python file as embedded file worked. In following PRs, will use a static file for dev/build tooling and then read it at runtime and inject it into an embedded file.